### PR TITLE
ipv6 support

### PR DESCRIPTION
**Pull request by volth**
support for IPv6 addresses

### CODE REVIEW

File: src/vmod_geoip.c [hunk 1]
```diff
@@ -17,6 +17,23 @@
...
 
+typedef struct vmod_geoip_db_type {
+    GeoIP    *ipv4;
+    GeoIP    *ipv6;
+} GeoipDB;
+
+static void
+cleanup_db(GeoipDB *db)
+{
+	if (db->ipv4)
```
Comment (gquintard) [inline src/vmod_geoip.c:28]: Why check since you assert at l61 and l65?

File: src/vmod_geoip.c [hunk 2]
```diff
@@ -17,6 +17,23 @@
 
 #include "vcc_if.h"
 
+typedef struct vmod_geoip_db_type {
```
Comment (gquintard) [inline src/vmod_geoip.c:20]: let's have a magic number here since it's now an object

File: src/vmod_geoip.c [hunk 3]
```diff
@@ -31,22 +48,55 @@ vmod_event(VRT_CTX, struct vmod_priv *pp, enum vcl_event_e evt)
...
+		AN(db->ipv6);
+		GeoIP_set_charset(db->ipv6, GEOIP_CHARSET_UTF8);
 	}
 
 	return (0);
 }
 
 static const char *
-vmod_region_name_by_addr(GeoIP *gi, const char *ip)
+vmod_country_code_by_addr(GeoipDB *gi, const char *ip)
```
Comment (gquintard) [inline src/vmod_geoip.c:73]: would have been good to split the ipv6 commit and the new APIs one

File: src/vmod_geoip.c [hunk 4]
```diff
@@ -70,7 +120,7 @@ vmod_region_name_by_addr(GeoIP *gi, const char *ip)
 			str = "Unknown";				\
 		return (str);						\
 	}
-GEOIP_PROPERTY(country_code, GeoIP_country_code_by_addr);
-GEOIP_PROPERTY(country_name, GeoIP_country_name_by_addr);
+GEOIP_PROPERTY(country_code, vmod_country_code_by_addr);
```
Comment (gquintard) [inline src/vmod_geoip.c:123]: need to update the vcc too, don't you?

File: src/vmod_geoip.c [hunk 5]
```diff
@@ -31,22 +48,55 @@ vmod_event(VRT_CTX, struct vmod_priv *pp, enum vcl_event_e evt)
...
 	}
 
 	return (0);
 }
 
 static const char *
-vmod_region_name_by_addr(GeoIP *gi, const char *ip)
+vmod_country_code_by_addr(GeoipDB *gi, const char *ip)
+{
+	if (strchr(ip, ':') != NULL)
```
Comment (gquintard) [inline src/vmod_geoip.c:75]: need to check in can ip == NULL

File: src/vmod_geoip.c [hunk 6]
```diff
@@ -31,22 +48,55 @@ vmod_event(VRT_CTX, struct vmod_priv *pp, enum vcl_event_e evt)
 		 * adding * the flag "MAP_32BIT" to the mmap call. MMAP is not
 		 * avail for WIN32.
 		 */
-		pp->priv = GeoIP_new(GEOIP_MMAP_CACHE);
+		pp->priv = malloc(sizeof(GeoipDB));
 		AN(pp->priv);
-		pp->free = (vmod_priv_free_f *)GeoIP_delete;
-		GeoIP_set_charset((GeoIP *)pp->priv, GEOIP_CHARSET_UTF8);
+
+		GeoipDB *db = (GeoipDB *)pp->priv;
```
Comment (gquintard) [inline src/vmod_geoip.c:54]: use miniobj.h to cast

File: src/vmod_geoip.c [hunk 7]
```diff
@@ -31,22 +48,55 @@ vmod_event(VRT_CTX, struct vmod_priv *pp, enum vcl_event_e evt)
...
+vmod_country_code_by_addr(GeoipDB *gi, const char *ip)
+{
+	if (strchr(ip, ':') != NULL)
+		return GeoIP_country_code_by_addr_v6(gi->ipv6, ip);
+	else
+		return GeoIP_country_code_by_addr(gi->ipv4, ip);
+}
+
+static const char *
+vmod_country_name_by_addr(GeoipDB *gi, const char *ip)
```
Comment (gquintard) [inline src/vmod_geoip.c:82]: miniobj.h assert to check non-null and correct magic

File: src/vmod_geoip.c (hunk 4 above)
Comment (volth) [inline src/vmod_geoip.c]: what to update in the vcc?

Comment (gquintard) [reply src/vmod_geoip.c:123]: ah, my bad, the API didn't change.

**Comment by gquintard** [comment]
What's up with this one?

**Comment by Rod-Gomes** [comment]
Could anyone take a look at this? It would be helpful to have IPv6 support.

**Comment by gquintard** [comment]
if anyone wants to do the final touchups, I think that can go in. Not sure why @volth closed it.

**Pull request closed without merging**